src/main/java/org/spongepowered/mod/mixin/core/entity/player/MixinEntityPlayer.java

```diff
@@ -50,4 +67,48 @@ public EntityItem dropOneItem(boolean dropAll) {
     public EntityItem dropPlayerItemWithRandomChoice(ItemStack itemStackIn, boolean unused) {
         return this.dropItem(itemStackIn, false, false);
     }
+
+    @Overwrite
```

Comment:
This should be moved to SpongeCommon. As for the PlayerWakeupEvent, just create a mixin to fire it before line 94.